packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -147,6 +147,23 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  // Can't upgrade if Expo is running
+  let status = await Project.currentStatus(projectRoot);
+  if (status === 'running') {
+    let answer = await prompt({
+      type: 'confirm',
+      name: 'closeExpo',
+      message: 'Expo needs to close before we can upgrade your dependencies. Do you want to close it?',
```

Comment:
```suggestion
      message: 'We found an existing expo-cli instance running for this project, it needs to be closed to continue.',
```

maybe something like this and then present two options:

```
- Close it for me and continue upgrade
- Cancel, I'm not ready yet
```

tbh i'm not sure how important this is though.. we could just close it for people and tell them that we closed it. it's not a destructive action, it costs nothing for them to close and reopen the packager.

Reply:
I don't think a lot of devs will even encounter this. Maybe just telling them that we closed it is sufficient and leaves some extra room for other questions it might encounter in the future 😄

Reply:
sounds good

Reply:
approved so we can ship it once you have made this change!

Reply:
Done! 😄